src/instance-group-manager.js

```diff
@@ -0,0 +1,550 @@
+InstanceGroupManager.prototype.getVMsStream = common.paginator.streamify(
+  'getVMs'
+);
+
+/**
+ * Remove the VMs managed by this instance group.
+ *
+ * @see [InstanceGroupManagers: abandonInstances API Documentation]{@link https://cloud.google.com/compute/docs/reference/v1/instanceGroupManagers/abandonInstances}
+ *
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this request.
+ * @param {Operation} callback.operation - An operation object
+ *     that can be used to check the status of the request.
+ * @param {object} callback.apiResponse - The full API response.
+ *
+ * @example
+ * const Compute = require('@google-cloud/compute');
+ * const compute = new Compute();
+ * const zone = compute.zone('us-central1-a');
+ * const instanceGroupManager = zone.instanceGroupManager('web-servers');
+ *
+ * instanceGroupManager.removeVMs(function(err, operation, apiResponse) {
+ *   // `operation` is an Operation object that can be used to check the status
+ *   // of the request.
+ * });
+ *
+ * //-
+ * // If the callback is omitted, we'll return a Promise.
+ * //-
+ * instanceGroupManager.removeVMs().then(function(data) {
+ *   const operation = data[0];
+ *   const apiResponse = data[1];
+ * });
+ */
+InstanceGroupManager.prototype.removeVMs = function(callback) {
+  var self = this;
+
+  this.request(
+    {
+      method: 'POST',
+      uri: '/abandonInstances',
+    },
+    function(err, resp) {
+      if (err) {
+        callback(err, null, resp);
+        return;
+      }
+
+      var operation = self.zone.operation(resp.name);
+      operation.metadata = resp;
+
+      callback(err, operation, resp);
+    }
+  );
+};
+
+/**
+ * Recreate the VMs managed by this instance group.
+ *
+ * @see [InstanceGroupManagers: recreateInstances API Documentation]{@link https://cloud.google.com/compute/docs/reference/v1/instanceGroupManagers/recreateInstances}
+ *
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this request.
+ * @param {Operation} callback.operation - An operation object
+ *     that can be used to check the status of the request.
+ * @param {object} callback.apiResponse - The full API response.
+ *
+ * @example
+ * const Compute = require('@google-cloud/compute');
+ * const compute = new Compute();
+ * const zone = compute.zone('us-central1-a');
+ * const instanceGroupManager = zone.instanceGroupManager('web-servers');
+ *
+ * instanceGroupManager.recreateVMs(function(err, operation, apiResponse) {
+ *   // `operation` is an Operation object that can be used to check the status
+ *   // of the request.
+ * });
+ *
+ * //-
+ * // If the callback is omitted, we'll return a Promise.
+ * //-
+ * instanceGroupManager.recreateVMs().then(function(data) {
+ *   const operation = data[0];
+ *   const apiResponse = data[1];
+ * });
+ */
+InstanceGroupManager.prototype.recreateVMs = function(callback) {
```

Comment:
Should we accept the `instances` part of the request body: https://cloud.google.com/compute/docs/reference/rest/v1/instanceGroupManagers/recreateInstances?

instances[] | string | The URLs of one or more instances to recreate. This can be a full URL or a partial URL, such as zones/[ZONE]/instances/[INSTANCE_NAME].
-- | -- | --